Tests for deleting tasks and toggling view of completed tasks

OrianP opened the pull request:
Hi Milly, 

This is a pull request draft as I am not yet ready to merge because I would like to refactor the code a little before merging. 
I added the two tests and a few mock elements in the `index.html` file just to demonstrate the test failing. I had to make a delete button otherwise when I try to click it in the test I will get en error that it cannot click `null`. These are just placeholder elements based on what I imagined the real application would do. We will delete them once we write the actual app logic which I imagined would go something like this:

When submit button is clicked:
// create a new `li` 
// create a new delete `button`
// append the delete `button` to the `li`
// update the text content of the `li` to the value of `input`
// append the `li` to `ul`

As for the filter button, I thought we would have a `".completed-task"` class list for completed tasks and that when a user clicks the filter button it toggles the `hidden` attribute to hide/reveal the completed task. It will make sense when you see what I wrote in the test 😄 

Let me know if you have any comments on the code for now and I will update the pull request once I refactor and am ready to merge. I'll try to get this done by Monday.

OrianP committed:
Add test for deleting item from list
add mock html elements to demonstrate test failure

OrianP committed:
Add test for toggling a filter button to hide completed tasks
code to be refactored

millipede-cpu reviewed (commented):
Very organised pseudocode, clearly states the setup of the test and how it will be refactored once the app files are added.  The filter button functionality is great, but possibly could be renamed to secret button.

millipede-cpu reviewed (commented):
Hi Orian, great start on structuring the tests together. I like the use of `childElementCount` in the delete button so that when a user deletes an item, it will update the to-do list automatically.

OrianP committed:
Refactor
change li element in task list to input element in index.html and refactor tests accordingly

millipede-cpu reviewed (approved):
Hi Orian, thank you for refactoring, it looks good, I'm happy to merge 👍🏼
